### PR TITLE
fix(Player/Groups): Level restriction on allowing raid groups

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  And it's configurable, pog. 
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/12107

## Tests Performed:

- tested ingame
- 

## How to Test the Changes:

1. Try to create a raid group with a member below level 10

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by temperrr** [comment]
Will test tomorrow

**Comment by temperrr** [comment]
Tested, works as described
config works as described.

File: src/server/game/Groups/Group.cpp [hunk 1]
```diff
@@ -278,6 +279,16 @@ void Group::ConvertToLFG(bool restricted /*= true*/)
     SendUpdate();
 }
 
+bool Group::CheckLevelForRaid()
+{
+    for (member_citerator citr = m_memberSlots.begin(); citr != m_memberSlots.end(); ++citr)
+        if (Player* player = ObjectAccessor::FindPlayer(citr->guid))
```
Comment (Nyeriah) [inline src/server/game/Groups/Group.cpp:285]: FindConnectedPlayer is probably better. Iirc findplayer only checks the current map storage and only accounts for players currently in world (so no loading screen while teleporting from one place to another, for example)

Comment (Kitzunu) [reply src/server/game/Groups/Group.cpp:285]: Used the same as Group::ConvertToRaid uses. But I can redo

Comment (Nyeriah) [reply src/server/game/Groups/Group.cpp:285]: probably fine then

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**